### ISSUE TITLE
BigNumeric not usable after BigNumericUDT.deserialize

### ISSUE THREAD

**Issue by ghost** [GoogleCloudDataproc/spark-bigquery-connector#743]
As in, you can't actually do any useful operations on a `BigNumeric` on the Python side.

The simpliest workaround is to make `BigNumeric` a subclass of `Decimal`.

```python
gary@cluster-9629-m:/opt/conda/default/lib/python3.8/site-packages/google/cloud$ cat spark/bigquery/big_numeric_support.py
from pyspark.sql.types import UserDefinedType, StringType
from decimal import *

class BigNumericUDT(UserDefinedType):
    """
    SQL user-defined type (UDT) for BigNumeric.
    """

    @classmethod
    def sqlType(cls):
        return StringType()

    @classmethod
    def module(cls):
        return "google.cloud.spark.bigquery.big_numeric_support"

    @classmethod
    def scalaUDT(cls):
        return "org.apache.spark.bigquery.BigNumericUDT"

    def serialize(self, obj):
        if isinstance(obj, BigNumeric):
            return str(obj)
        else:
            raise TypeError("cannot serialize %r of type %r" % (obj, type(obj)))

    def deserialize(self, datum):
        return BigNumeric(str(datum))

    def simpleString(self):
        return "BigNumeric"


class BigNumeric(Decimal): pass

    #def __init__(self, number):
    #    if isinstance(number, Decimal):
    #        self.number = number
    #    else:
    #        raise TypeError("BigNumeric Constuctor: expecting Decimal but got %r " % type(number))
```

**Comment by gary-arcana** [comment]
For my specific use case, I need `__str__` to return `str(big_num.number)`

**Comment by davidrabinowitz** [comment]
BigNumeric handling has been fixed in version 0.31.0
